Sources/Moya/Plugins/AccessTokenPlugin.swift

```diff
@@ -51,12 +54,20 @@ public struct AccessTokenPlugin: PluginType {
      - returns: The modified `URLRequest`.
     */
     public func prepare(_ request: URLRequest, target: TargetType) -> URLRequest {
-        if let authorizable = target as? AccessTokenAuthorizable, authorizable.shouldAuthorize == false {
-            return request
-        }
+        guard let authorizable = target as? AccessTokenAuthorizable else { return request }
+
+        let authorizationType = authorizable.authorizationType
 
         var request = request
-        request.addValue(authVal, forHTTPHeaderField: "Authorization")
+
+        switch authorizationType {
+        case .basic:
```

Comment:
Since `.basic` and `.bearer` have the same implementation, maybe it'd be better to do `case .basic, .bearer:`?